File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/define/mat_live_room_audio_seat.dart
import 'package:flutter/material.dart';
import 'package:matlive_client_sdk_flutter/audio/define/mat_live_user.dart';

class MatLiveRoomAudioSeat {
  int seatIndex = 0;
  int rowIndex = 0;
  int columnIndex = 0;
  GlobalKey seatKey = GlobalKey();
  ValueNotifier<MatLiveUser?> lastUser = ValueNotifier(null);
  ValueNotifier<MatLiveUser?> currentUser = ValueNotifier(null);
  ValueNotifier<bool> isLocked = ValueNotifier(false);

  MatLiveRoomAudioSeat(this.seatIndex, this.rowIndex, this.columnIndex);
}

File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/define/MatLiveRoomAudioSeat.kt
package com.matnsolutions.matlive_sdk.audio.define

import androidx.lifecycle.MutableLiveData

class MatLiveRoomAudioSeat(
    var seatIndex: Int,
    var rowIndex: Int,
    var columnIndex: Int
) {
    var seatKey: Any = Any() // Placeholder for GlobalKey equivalent
    var lastUser: MutableLiveData<MatLiveUser?> = MutableLiveData(null)
    var currentUser: MutableLiveData<MatLiveUser?> = MutableLiveData(null)
    var isLocked: MutableLiveData<Boolean> = MutableLiveData(false)
}